src/typed-method-types/workflows/triggers/event-data/message_posted.ts

```diff
@@ -0,0 +1,34 @@
+export const MessagePosted = {
+  /**
+   * A unique identifier for the app that posted the message. WARNING: Only available if posted by an app! If your message event trigger's filter matches a posted message, but it came from a non-app user, and your message event trigger uses this context data, the workflow execution will fail.
```

Comment:
> Missing trigger data based on context should, if I understand correctly, be handled in the future by this 'speedbump modal' mentioned in thread.

We can't open a speedbump modal for non-interactive triggers like `MessagePosted` though since there is no user to collect the missing properties from. Will DM you to figure out where to route this.

As far as the code, I think your "Only available when invoked from X" is good for now. We can then work on fixing the failure or getting it documented on the doc site with a workaround that we can later link to